index.js

```diff
@@ -1,94 +1,62 @@
 var vm = require('vm');
 var requestModule = require('request');
-var jar = requestModule.jar();
-
-var request      = requestModule.defaults({jar: jar}); // Cookies should be enabled
-var UserAgent    = 'Ubuntu Chromium/34.0.1847.116 Chrome/34.0.1847.116 Safari/537.36';
-var Timeout      = 6000; // Cloudflare requires a delay of 5 seconds, so wait for at least 6.
-var cloudscraper = {};
-var MaxChallengesToSolve = 3; // Support only this max challenges in row. If CF returns more, throw an error
-
-/**
- * Performs get request to url with headers.
- * @param  {String}    url
- * @param  {Function}  callback    function(error, response, body) {}
- * @param  {Object}    headers     Hash with headers, e.g. {'Referer': 'http://google.com', 'User-Agent': '...'}
- */
-cloudscraper.get = function(url, callback, headers) {
-  performRequest({
-    method: 'GET',
-    url: url,
-    headers: headers
-  }, callback);
-};
-
-/**
- * Performs post request to url with headers.
- * @param  {String}        url
- * @param  {String|Object} body        Will be passed as form data
- * @param  {Function}      callback    function(error, response, body) {}
- * @param  {Object}        headers     Hash with headers, e.g. {'Referer': 'http://google.com', 'User-Agent': '...'}
- */
-cloudscraper.post = function(url, body, callback, headers) {
-  var data = '';
-  var bodyType = Object.prototype.toString.call(body);
-
-  if(bodyType === '[object String]') {
-    data = body;
-  } else if (bodyType === '[object Object]') {
-    data = Object.keys(body).map(function(key) {
-      return key + '=' + body[key];
-    }).join('&');
-  }
-
-  headers = headers || {};
-  headers['Content-Type'] = headers['Content-Type'] || 'application/x-www-form-urlencoded; charset=UTF-8';
-  headers['Content-Length'] = headers['Content-Length'] || data.length;
 
-  performRequest({
-    method: 'POST',
-    body: data,
-    url: url,
-    headers: headers
-  }, callback);
+var originalDefaults = requestModule.defaults;
```

Comment:
Wow, this is very nice. I wasn't aware of this feature in request. Looks like it would make things much easier